test/e2e/scale.go

```diff
@@ -83,11 +87,19 @@ type scaleSpecInput struct {
 	// Can be overridden by variable CAPI_SCALE_CONTROLPLANE_MACHINE_COUNT.
 	ControlPlaneMachineCount *int64
 
-	// WorkerMachineCount defines number of worker machines to be added to each workload cluster.
+	// WorkerMachineCount defines number of worker machines per machine deployment of the workload cluster.
 	// If not specified, 1 will be used.
 	// Can be overridden by variable CAPI_SCALE_WORKER_MACHINE_COUNT.
 	WorkerMachineCount *int64
 
+	// MachineDeploymentCount defines the number of MachineDeployments to be used per workload cluster.
+	// If not specified, 1 will be used.
+	// Can be overridden by variable CAPI_SCALE_MACHINE_DEPLOYMENT_COUNT.
+	// Note: This assumes that the cluster template of the specified flavor has at least one machine deployment.
```

Comment:
WDYT about enforcing that the template should always have exactly 1 MD? I don't think it makes much sense to support anything else as when we enforce that 1 exists configuring MachineDeploymentCount always works. Otherwise it becomes a bit awkward